chore(step-generation): Update immer from 5.1.0 to 10.1.3

SyntaxColoring opened the pull request:
## Overview

This upgrades our Immer dependency from v5.1.0 (2019-12-20) to v10.1.3 (2025-09-01). So far, only step-generation uses it. Goes towards EXEC-2002.

## Test Plan and Hands on Testing

Just CI.

## Changelog

1. Upgrade Immer:
   1. Manually edit step-generation/package.json to update Immer's version range.
   1. ~~Run `yarn upgrade` at the top level. I tried `yarn workspace '@opentrons/step-generation' upgrade`, but that failed with an obscure error that looks like https://github.com/yarnpkg/yarn/issues/7734.~~ Run `make teardown-js && make setup-js`.
   1. Update step-generation's source. As far as I can tell, the only thing that needed to change was the import syntax.
1. Also delete a duplicate entry for `opentrons-ai-client` in our package.json.

## Review requests

* Is this the right way to upgrade a single project's JS dependency?
* Take a look through Immer's release notes and make sure I'm not missing some subtlety that would require more source code changes.
   * https://github.com/immerjs/immer/releases/tag/v6.0.0
   * https://github.com/immerjs/immer/releases/tag/v7.0.0
   * https://github.com/immerjs/immer/releases/tag/v8.0.0
   * https://github.com/immerjs/immer/releases/tag/v9.0.0
   * https://github.com/immerjs/immer/releases/tag/v10.0.0

## Risk assessment

Lowish. This is a big leap of an update, but we use Immer very lightly.

SyntaxColoring committed:
Delete duplicate opentrons-ai-client workspaces entry.

SyntaxColoring committed:
Edit step-generation/package.json to updgrade Immer's version spec.

SyntaxColoring committed:
Run `yarn upgrade` at the top level.

`yarn workspace '@opentrons/step-generation' upgrade` fails with an obscure error. Possibly https://github.com/yarnpkg/yarn/issues/7734.

SyntaxColoring committed:
Update step-generation source.

jerader commented on package.json:
```diff
@@ -12,12 +13,10 @@
-      "api-client",
-      "react-api-client",
-      "usb-bridge/node-client",
-      "opentrons-ai-client"
```
oof good catch

jerader reviewed (approved):
seems low risk enough to me!

SyntaxColoring committed:
Revert "Run `yarn upgrade` at the top level."

This reverts commit e7a8dbe27a63390150029dbe24157bf3dadd87df.

SyntaxColoring committed:
`make teardown-js && make setup-js`

I don't fully understand what's going on here, but apparently this command bumps immer and nothing else, whereas `yarn upgrade` bumped everything, and something inside the everything-bump made a bunch of type-checking fail.
